controllers/extensions/addon_controller_stages.go

```diff
@@ -161,6 +166,115 @@ type terminalStateStage struct {
 	stageCtx
 }
 
+func getKubeBlocksDeploy(r *AddonReconciler, ctx context.Context) (*v1.Deployment, error) {
+	deploys := &v1.DeploymentList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		"app.kubernetes.io/name":      "kubeblocks",
+		"app.kubernetes.io/component": "apps",
+	})
+	if err := r.Client.List(ctx, deploys, client.InNamespace(viper.GetString(constant.CfgKeyCtrlrMgrNS)),
+		client.MatchingLabelsSelector{Selector: labelSelector}); err != nil {
+		return nil, err
+	}
+	if deploys == nil || len(deploys.Items) == 0 {
```

Comment:
1. `if deploys.Items == nil || len(deploys.Items) == 0`
2. the number of KB deployment should be one. Any other value should return an error

Reply:
Thanks for all of your suggestions! 
I will take them into consideration